wls-common/testing/src/main/java/de/muenchen/oss/wahllokalsystem/wls/common/testing/archunit/condition/NestedTestsuitesHaveMatchingMethodCondition.java

```diff
@@ -0,0 +1,90 @@
+package de.muenchen.oss.wahllokalsystem.wls.common.testing.archunit.condition;
+
+import com.tngtech.archunit.core.domain.JavaClass;
+import com.tngtech.archunit.lang.ArchCondition;
+import com.tngtech.archunit.lang.ConditionEvents;
+import com.tngtech.archunit.lang.SimpleConditionEvent;
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.Set;
+import lombok.val;
+import org.junit.jupiter.api.Nested;
+import org.springframework.util.StringUtils;
+
+public class NestedTestsuitesHaveMatchingMethodCondition {
+
+    public static final ArchCondition<JavaClass> haveMatchingPublicMethodNameIfTheyAreHighestNestedClass = new ArchCondition<>(
+            "have matching public method name if they are highest nested class") {
+        @Override
+        public void check(JavaClass classWithNestedAnnotation, ConditionEvents events) {
+            var topNestedClass = getTopNestedClass(classWithNestedAnnotation);
+            val expectedMethodName = StringUtils.uncapitalize(topNestedClass.getSimpleName());
+            var topEnclosingClass = getTopEnclosingClass(classWithNestedAnnotation);
+
+            if (topEnclosingClass.isPresent()) {
+
+                // exclude specific testfiles from @nested naming convention
+                String classNameString = topEnclosingClass.map(JavaClass::getFullName).orElse("failed to extract class name");
```

Comment:
Durch die Prüfung zuvor ist bereits sichergestellt das `topEnclosingClass` nicht empty ist. Dadurch kannst du `get` verwenden und brauchst den Trick mit `orElse` nicht.

```suggestion
                val classNameString = topEnclosingClass.get().getFullName();
```

Reply:
[2722dd5](https://github.com/it-at-m/Wahllokalsystem/pull/918/commits/2722dd5930941e41c043042d42683622957f37d0)